climada_petals/hazard/tc_rainfield.py

```diff
@@ -66,204 +208,1449 @@ def set_from_tracks(self, *args, **kwargs):
         self.__dict__ = TCRain.from_tracks(*args, **kwargs).__dict__
 
     @classmethod
-    def from_tracks(cls, tracks, centroids=None, dist_degree=3, description='', pool=None,
-                    intensity_thres=None):
-        """Computes rainfield from tracks based on the RCLIPER model.
-        Parallel process.
+    def from_tracks(
+        cls,
+        tracks: TCTracks,
+        centroids: Optional[Centroids] = None,
+        pool: Optional[pathos.pools.ProcessPool] = None,
+        description: str = '',
+        model: str = 'R-CLIPER',
+        model_kwargs: Optional[dict] = None,
+        ignore_distance_to_coast: bool = False,
+        store_rainrates: bool = False,
+        metric: str = "equirect",
+        intensity_thres: float = DEF_INTENSITY_THRES,
+        max_latitude: float = 61,
+        max_dist_inland_km: float = 1000,
+        max_dist_eye_km: float = DEF_MAX_DIST_EYE_KM,
+        max_memory_gb: float = DEF_MAX_MEMORY_GB,
+    ):
+        """
+        Create new TCRain instance that contains rainfields from the specified tracks
+
+        This function sets the `intensity` attribute to contain, for each centroid,
+        the total amount of rain experienced over the whole period of each TC event in mm.
+        The amount of rain is set to 0 if it does not exceed the threshold `intensity_thres`.
+
+        The `category` attribute is set to the value of the `category`-attribute
+        of each of the given track data sets.
+
+        The `basin` attribute is set to the genesis basin for each event, which
+        is the first value of the `basin`-variable in each of the given track data sets.
+
+        Optionally, the time-dependent rain rates can be stored using the `store_rainrates`
+        function parameter (see below).
 
         Parameters
         ----------
-        tracks : TCTracks
-            tracks of events
+        tracks : climada.hazard.TCTracks
+            Tracks of storm events.
         centroids : Centroids, optional
-            Centroids where to model TC.
-            Default: global centroids.
-        disr_degree : int
-            distance (in degrees) from node within which
-            the rainfield is processed (default 3 deg,~300km)
-        description : str, optional
-            description of the events
+            Centroids where to model TC. Default: global centroids at 360 arc-seconds resolution.
         pool : pathos.pool, optional
-            Pool that will be used for parallel computation of wind fields. Default: None
+            Pool that will be used for parallel computation of rain fields. Default: None
+        description : str, optional
+            Description of the event set. Default: "".
+        model : str, optional
+            Parametric rain model to use: "R-CLIPER" (faster and requires less inputs, but
```

Comment:
I would suggest that references for each model are provided here as well. I see that they are provided in the individual method for each model, but for the normal user it might be useful to have them in the documentation of this main method.

Reply:
Thanks! I added more details about the models to the docstring of `TCRain.from_tracks`.